repos/system_upgrade/common/actors/scanfilestocopyintotargetuserspace/tests/test_scanfilestocopyintotargetuserspace.py

```diff
@@ -0,0 +1,77 @@
+import os
+
+import pytest
+
+from leapp.libraries.actor.scanfilestocopyintotargetuserspace import scan_files_to_copy
+from leapp.libraries.common.testutils import produce_mocked
+from leapp.libraries.stdlib import api
+
+
+@pytest.fixture
+def isfile_default_config():
+    config = {
+        '/etc/hosts': True
+    }
+    return config
+
+
+def do_files_to_copy_contain_entry_with_src(files_to_copy, src):
+    """Searches the files to be copied for an entry with src field that matches the given src."""
+    is_present = False
+    for file_to_copy in files_to_copy:
+        if file_to_copy.src == src:
+            is_present = True
+            break
+    return is_present
+
+
+def make_mocked_isfile(configuration):
+    """
+    Creates mocked isfile function that returns values according the given configuration.
+
+    The created function raises :class:`ValueError` sould the unit under test try to isfile
+    a path that is not present in the configuration.
+
+    One global mocked function with configuration is error prone as individual test would
+    have to return the configuration into the original state after execution.
+    """
+
+    def mocked_isfile(path):
+        if path in configuration:
+            return configuration[path]
+        error_msg = 'The actor tried to isfile a path that it should not. (path `{0}`)'
+        raise ValueError(error_msg.format(path))
+    return mocked_isfile
+
+
+def test_etc_hosts_present(monkeypatch, isfile_default_config):
+    """Tests whether /etc/hosts is not identified to be copied into target userspace when it is missing."""
+    mocked_isfile = make_mocked_isfile(isfile_default_config)
+    actor_products = produce_mocked()
+    monkeypatch.setattr(os.path, 'isfile', mocked_isfile)
+    monkeypatch.setattr(api, 'produce', actor_products)
+
+    scan_files_to_copy()
+
+    fail_msg = 'Didn\'t indentify any files to copy into target userspace (at least /etc/hosts shoud be)'
+    assert actor_products.model_instances, fail_msg
+
+    files_to_copy = actor_products.model_instances
+    should_copy_hostsfile = do_files_to_copy_contain_entry_with_src(files_to_copy, '/etc/hosts')
+    assert should_copy_hostsfile, 'Failed to identify /etc/hosts as a file to be copied into target userspace.'
+
+
+def test_etc_hosts_missing(monkeypatch, isfile_default_config):
+    """Tests whether /etc/hosts is identified to be copied into target userspace when it is present."""
+    isfile_default_config['/etc/hosts'] = False  # The file is not present or is a directory (-> shoud not be copied)
+    mocked_isfile = make_mocked_isfile(isfile_default_config)
+    actor_products = produce_mocked()
```

Comment:
```suggestion
    actor_produces = produce_mocked()
```

Reply:
Suggested change applied in the latest version.